infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -69,6 +70,21 @@
 
     private Orchestrator orchestrator;
 
+    /**
+     * System down handler to break out of live-locks if the runtime cannot reach the cluster for a
+     * certain amount of time.
+     */
+    private final Runnable runtimeSystemDownHandler = () -> {
+        log.warn("ManagementServer: Runtime stalled. Invoking systemDownHandler.");
+        throw new StalledRuntimeException();
+    };
+
+    /**
+     * The number of retries to be made to execute any RPC request before the runtime gives up and
+     * invokes the systemDownHandler.
+     */
+    private static final int SYSTEM_DOWN_HANDLER_TRIGGER_LIMIT = 50;
```

Comment:
why 50? Maybe add the rationale in a comment

Reply:
I'll add the comment.